client/tooltool.py

```diff
@@ -821,9 +821,12 @@ def touch(f):
 def request(url, auth_file=None):
     req = Request(url)
     _authorize(req, auth_file)
-    with closing(urllib2.urlopen(req)) as f:
-        log.debug("opened %s for reading" % url)
-        yield f
+    try:
```

Comment:
I don't think this block is actually retrying? It's just eating the exception, and then returning. I think you need to stick the try/except into a loop, and break after a few attempts.

Reply:
This is not the actualy retry, this is meant to prevent the code throwing the exception and make the job look like it failed when there was more that one request attempt.
I don't mind merging the commit without this `try` block, but it will not fix the issue of the linked bug.

Reply:
As we've discussed elsewhere, eating the exception here prevents the retry from happening. I noticed in some of your tests that in some cases, something is retrying the overall call to `tooltool.py`, which is why you're seeing retries in those tests -- but in the current state of this code I don't see any way in which retries here work. To have `tooltool.py` retry we either need to:
1) Stop eating exceptions in this block. This will let `retriable` do the retries, but exceptions will be printed.
or
2) Get rid of `retriable`, and manually handle the retries in this function. This will let us do whatever logging we need.